README.md

```diff
@@ -35,7 +31,7 @@ make test
 
 To run the end to end test suite, you need to have the Mattermost server project downloaded and configured in your system. Check the [Developer Setup](https://developers.mattermost.com/contribute/server/developer-setup/) guide on how to configure a local server instance. The tests will search for a `mattermost-server` folder in the same directory where the `mattermost-plugin-apps` is.
 
-With the `mattermost-server` folder present, the only thing that needs to be done before running the tests themselves is to start the Mattermost docker development environment. The environment only needs to be started once, and then the tests can run as many times as needed. To start the docker environment, change to the `mattermost-server` project directory and run:
+With the `mattermost-server` folder present, the only thing that needs to be done before running the tests themselves is to start the Mattermost Docker development environment. The environment only needs to be started once, and then the tests can run as many times as needed. To start the Docker environment, change to the `mattermost-server` project directory and run:
```

Comment:
I didn't realize we had this excerpt about the "Mattermost Docker development environment". We need to make sure it's known that it's different than the "App Framework Docker development environment" described above.

Reply:
It seems like that's something that could trip users up. Initially I thought a simple "This is not the same as the App Framework Docker development preview" would suffice but I think we need more context.

I'm thinking something like: "Separate to the App Framework Docker development preview environment, you can also run tests using the Mattermost Docker development environment..." 

Do you have any additional info I can add to that to explain the difference and any potential conflicts?

Reply:
The "Mattermost Docker development environment" is used to spin up several Docker containers, so that a Mattermost server can communicate with those containers. When you are making changes to  Mattermost projects (e.g. server, webapp), you have to manually run the Mattermost server, which communicates to these Docker containers.

The App Framework e2e tests (which is what this section of the README is referring to) in the App Framework project require these same containers to be running. However these tests don't need a Mattermost server to be running. The tests instead sort of mimic the behavior of a running server using shared code of the mattermost-server repository. It sort of has a "fake server" running, completely separate from the running containers, but communicating with the containers.

When you're developing your own App, you need an actual Mattermost server to be running. The "Mattermost App Framework Docker development environment" helps accomplish this by setting up a minimalistic environment with just two containers. One is for the database Mattermost communicates with, and the other container runs the actual Mattermost server. The other containers present in the "Mattermost Docker development environment" are unnecessary for the purposes of build Apps. So the advantage here is that there is just one dependency to start developing Apps. Just spin up the "Mattermost App Framework Docker development environment"!

Some differences between the environments:

- The App Framework dev environment (maybe this should be the name? It's shorter) has its own Mattermost server and it's fully configured to start developing Apps. The config values are set correctly so no modifications need to be done there.

- The App Framework dev environment also has a starter App built-in as a third container, but this can be ignored if the developer wishes to run their App outside of the dev environment, while still using it by communicating with it from outside of the containers.

- The App Framework e2e tests cannot be run with the App Framework dev environment, but that is irrelevant if you are trying to create an App. I think there's a chance for miscommunication because we are trying to speak to two different audiences here. The e2e tests portion should be de-emphasized compared to the App development side. It's much more likely someone reading this README is developing an App, as opposed to editing or running e2e tests locally.

More specific information about the App Framework Docker development environment is explained in [dev/README.md](https://github.com/mattermost/mattermost-plugin-apps/blob/master/dev/README.md).

Also, now that I think about it, you may just need a database running to run the e2e tests, so maybe we can use that environment for both things. That's a totally other topic though 🙂

Reply:
Why is the Apps Framework development environment even mentioned in the "End to end tests" section? For now, it has nothing to do with it.